src/components/smart/Table.vue

```diff
@@ -1,69 +1,321 @@
+        <input
+          v-model="searchQuery"
+          class="h-full w-full bg-primary py-3"
+          :placeholder="searchBar.placeholder ?? 'Search...'"
+        />
+      </div>
+      <div v-if="isSpinnerEnabled" class="mx-auto my-3 h-5 w-5 text-center">
+        <HoppSmartSpinner />
+      </div>
+
+      <table v-else-if="list" class="w-full">
+        <thead v-if="list.length > 0">
           <tr
-            class="text-sm text-left border-b border-dividerDark bg-primaryLight text-secondary"
+            class="border-b border-dividerDark bg-primaryLight text-left text-sm text-secondary"
           >
-            <th v-for="th in headings" scope="col" class="px-6 py-3">
-              {{ th.label ?? th.key }}
+            <th v-if="checkbox" class="px-3">
+              <div class="flex h-full items-center justify-center">
+                <input
+                  ref="selectAllCheckbox"
+                  type="checkbox"
+                  :checked="areAllRowsSelected"
+                  class="h-full w-full"
+                  @click.stop="toggleAllRows"
+                />
+              </div>
             </th>
+            <slot name="head">
+              <th v-for="th in headings" scope="col" class="px-6 py-3">
+                {{ th.label ?? th.key }}
+              </th>
+            </slot>
           </tr>
-        </slot>
-      </thead>
+        </thead>
 
-      <tbody class="divide-y divide-divider">
-        <!-- We are using slot props for future proofing so that in future, we can implement features like filtering -->
-        <slot name="body" :list="list">
+        <tbody class="divide-y divide-divider">
           <tr
-            v-for="(rowData, rowIndex) in list"
+            v-for="(rowData, rowIndex) in workingList"
             :key="rowIndex"
             class="rounded-xl text-secondaryDark hover:cursor-pointer hover:bg-divider"
             :class="{ 'divide-x divide-divider': showYBorder }"
+            @click="onRowClicked(rowData)"
           >
-            <td
-              v-for="cellHeading in headings"
-              :key="cellHeading.key"
-              @click="!cellHeading.preventClick && onRowClicked(rowData)"
-              class="max-w-[10rem] py-1 pl-6"
-            >
-              <!-- Dynamic column slot -->
-              <slot :name="cellHeading.key" :item="rowData">
-                <!-- Generic implementation of the column -->
-                <div class="flex flex-col truncate">
-                  <span class="truncate">
-                    {{ rowData[cellHeading.key] ?? "-" }}
-                  </span>
-                </div>
-              </slot>
+            <td v-if="checkbox" class="px-3">
+              <div class="flex h-full items-center justify-center">
+                <input
+                  type="checkbox"
+                  :checked="isRowSelected(rowData)"
+                  @click.stop="toggleRow(rowData)"
+                />
+              </div>
```

Comment:
Is the `div` wrapper necessary here?

```suggestion
              <input
                type="checkbox"
                :checked="isRowSelected(rowData)"
                class="flex h-full w-full items-center justify-center"
                @click.stop="toggleRow(rowData)"
              />
```